staging/src/k8s.io/apiserver/pkg/server/lifecycle_signals.go

```diff
@@ -188,3 +203,11 @@ func (e *namedChannelWrapper) Signaled() <-chan struct{} {
 func (e *namedChannelWrapper) Name() string {
 	return e.name
 }
+
+func (e *namedChannelWrapper) SignaledAt() (time.Time, bool) {
+	value := e.signaledAt.Load()
+	if value == nil {
+		return time.Time{}, false
```

Comment:
so every time we don't have a value (most of the time), we're allocating?  Why not make a pointer and return nil to indicate "not signaled"?

Reply:
This is the pattern of use:
```
<-Signaledd() 

// the even has been signaled, we want to know when it was signaled
at, ok := SignaledAt()
```

I thought `time.Time, bool` or `time.Time, error` is more idiomatic, but I don't have any strong opinion here.

Reply:
I would side with @deads2k on this one. I would agree that `time.Time, bool` is clearer, but having to allocate a time object every time is not really necessary.

I would personally prefer having the original time object stored as an atomic pointer and returned as a Time pointer here.